cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -143,6 +143,7 @@ public void configure() {
     properties.put(Constants.PIPELINEID, GSON.toJson(pipeline));
     properties.put(Constants.STAGE_LOGGING_ENABLED, String.valueOf(config.isStageLoggingEnabled()));
     setProperties(properties);
+    setDriverResources(config.getDriverResources());
```

Comment:
ah good catch, this needs a null check too.  The reason is that the configs are not created programmatically, but are created by deserializing whatever the user has sent in the REST call.  So this could be null even if there is no way for it to be null by creating one programmatically.